### ISSUE TITLE
demo for two separate client

### ISSUE THREAD

**Issue by godomainz** [privacyresearchgroup/libsignal-typescript-demo#7]
current implementation is both clients in same app ?
but how to implement this in if two clients in separate apps ?
example would be client-server architecture ?
how to implement the same if "Adalheid is the client" and  "Brünhild is the server" ?

**Comment by rolfeschmidt** [comment]
Yes, for this demo we put both clients in the same app and simulate message sending with the [`sendMessage` function](https://github.com/privacyresearchgroup/libsignal-typescript-demo/blob/master/src/App.tsx#L139). We want the demo to show how to use the SDK and didn't want the complexity of a real messaging system to get in the way.

But of course to really use this you need messaging. So a couple of comments:
* For a client-server connection I would look at a (Noise Protocol)[http://www.noiseprotocol.org/] first. It is probably better suited, but may not be easy to get started with in Javascript (something we should improve, lmk if that would be of interest...)
* When we use this SDK internally, we use a hybrid websocket/REST backend to deliver messages. When an app comes online it calls the REST API to pull messages, then listens on the websocket. I'll try to get a quick demo service public for your reference later this week, but it will be AWS-specific (dynamo db, lambda, API Gateway) for ease of deployment. *It will not scale well and will not hide information about who converses with whom!*
* But the beauty of protocols like Signal is that we can use any message delivery system that meets our needs.

**tl;dr** A two-client demo with a backend is out of scope for this little project, but we do need a better explanation of how to set that up. Let's keep this issue open until we have that explanation along with a simple reference backend.

**Comment by godomainz** [comment]
@rolfeschmidt thanks. It would be wonderful if we can have a simple two-client demo.

**Comment by rolfeschmidt** [comment]
Took a little longer than I expected, but hopefully the app in this PR will be helpful for you.
